### PR TITLE
adding relation controller to user for clusters for search box

### PR DESCRIPTION
**Pull request by djudzinski**

### CODE REVIEW

File: eventhandlers/RainlabUserHandler.php [hunk 1]
```diff
@@ -84,18 +83,21 @@ public function addMethodsToUser($event)
...
+                return $model->surname.' '.$model->name;
+            });
         });
     }
 
     public function addFullNameColumn($event)
     {
         $event->listen('backend.list.extendColumns', function ($widget) {
-            if ($widget->getController() instanceof Users) {
+            if ($widget->getController() instanceof Users && $widget->model instanceof User) {
```
Comment (dzapek) [inline eventhandlers/RainlabUserHandler.php:96]: check this

**Review by dzapek — CHANGES_REQUESTED** [review]

**Review by dzapek — APPROVED** [review]

File: eventhandlers/RainlabUserHandler.php [hunk 2]
```diff
@@ -84,18 +83,21 @@ public function addMethodsToUser($event)
             $model->addDynamicMethod('canEnter', function ($cluster) use ($model) {
                 return $model->clusters()->whereSlug($cluster->slug)->first() ? true : false;
             });
+
+            $model->addDynamicMethod('getFullNameAttribute', function ($user) use ($model) {
+                return $model->surname . ' ' . $model->name;
```
Comment (tomaszstrojny) [inline eventhandlers/RainlabUserHandler.php:88]: Full name should be in `first name last name` format.

File: models/cluster/related_columns.yaml [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+        label: 'initbiz.cumuluscore::lang.cluster.name'
+        searchable: true
+        sortable: true
+    slug:
+        label: 'initbiz.cumuluscore::lang.cluster.slug'
+        searchable: true
+        sortable: true
+    plan[name]:
+        label: 'initbiz.cumuluscore::lang.cluster.plan'
+        searchable: true
```
Comment (tomaszstrojny) [inline models/cluster/related_columns.yaml:16]: Are you sure that `searchable` and `sortable` will work for related fields?

**Review by tomaszstrojny — COMMENTED** [review]
Looks good, although take a look at those two comments.

**Pull request closed without merging**